manifests/client.pp

```diff
@@ -77,9 +77,12 @@
       ensure => $package_ensure,
       name   => $package_name,
     }
-    $req = Package['snmp-client']
-  } else {
-    $req = undef
+  }
+
+  $req = $::osfamily ? {
+    'RedHat'             => [Package[$snmp::params::package_name], Package['snmp-client']],
```

Comment:
snmp::client inherits snmp::params so it shouldn't be necessary to include ::snmp at all.